<!--|
    Copyright (C) 2019 Learning Algorithms and Systems Laboratory, EPFL, Switzerland
    Authors:  Konstantinos Chatzilygeroudis (maintainer)
              Bernardo Fichera
              Walid Amanhoud
    email:   konstantinos.chatzilygeroudis@epfl.ch
             bernardo.fichera@epfl.ch
             walid.amanhoud@epfl.ch
    website: lasa.epfl.ch

    This file is part of iiwa_ros.

    iiwa_ros is free software: you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation, either version 3 of the License, or
    (at your option) any later version.

    iiwa_ros is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.

|-->
<?xml version="1.0"?>
<launch>
  <arg name="hardware_interface" default="EffortJointInterface"/>
  <arg name="robot_name" default="iiwa"/>
  <arg name="type" default="14" />

  <!-- Upload iiwa URDF -->
  <include file="$(find iiwa_description)/launch/iiwa$(arg type)_upload.launch">
    <arg name="hardware_interface" value="$(arg hardware_interface)"/>
    <arg name="robot_name" value="$(arg robot_name)"/>
  </include>

  <!-- Load iiwa service -->
  <include file="$(find iiwa_tools)/launch/iiwa_service.launch"/>

  <!-- call iiwa FRI driver -->
  <node pkg="iiwa_driver" type="iiwa_driver" name="iiwa_driver" respawn="false" output="screen">
    <remap from="/joint_states" to="/iiwa/joint_states"/>
    <remap from="/controller_manager" to="/iiwa/controller_manager"/>
    <!-- Load configurations from YAML file to parameter server -->
    <rosparam file="$(find iiwa_driver)/config/iiwa.yaml" command="load"/>
  </node>

  <!-- Spawn position controller -->
  <include file="$(find iiwa_control)/launch/iiwa_torque_position_control.launch"/>

</launch>